src/components/application_manager/src/request_controller.cc

```diff
@@ -364,9 +369,27 @@ void RequestController::onTimer() {
   SDL_AUTO_TRACE();
   SDL_DEBUG(
       "ENTER Waiting fore response count: " << waiting_for_response_.Size());
-  RequestInfoPtr probably_expired =
-      waiting_for_response_.FrontWithNotNullTimeout();
-  while (probably_expired && probably_expired->isExpired()) {
+  while(!stop_flag_) {
```

Comment:
@BrandonHe I am trying to understand new flow. I can see that you mixed timers and loops. There is simplified current logic of RequestCtrl [sequence](https://www.websequencediagrams.com/cgi-bin/cdraw?lz=dGl0bGUgUmVxdWVzdCBjb250cm9sbGVyCgoKQU0tPgASB0N0cmw6IEFkZAAiCAoADgsAFw9QdXQgcgBJB2luIHF1ZXVlIAAiDlRpbWVyOiBVcGRhdGUgdGltZXIKAA8FAAUWAA18RG8gbm90IHUAgRgTAIIPDU9uVGltZW91AIF9HXJvY2VzcyBmaXJzAIISEgCBZj0APVYgKGxhc3Qgb25lKQCDLBVTdXNwZW5kAIMtFQAODgo&s=earth) Could you create sequence diagram with new logic?